File: LMIS/LMIS.Web/Modules/InventoryManage/UpBookshelf/UpBookshelfColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.InventoryManage.Columns;

[ColumnsScript("InventoryManage.UpBookshelf")]
[BasedOnRow(typeof(UpBookshelfRow), CheckNames = true)]
public class UpBookshelfColumns
{
    [AlignRight]
    public long UpBookshelfId { get; set; }
    public string BookName { get; set; }
    public string BookshelfLocation { get; set; }
    public long Inventory { get; set; }
    public string OperateUserName { get; set; }
    [Width(200)]
    public string Remark { get; set; }
    
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime CreateTime { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime UpdateTime { get; set; }
}

File: LMIS/LMIS.Web/Modules/ServerTypes/InventoryManage/UpBookshelfColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { UpBookshelfRow } from "./UpBookshelfRow";

export interface UpBookshelfColumns {
    UpBookshelfId: Column<UpBookshelfRow>;
    BookName: Column<UpBookshelfRow>;
    BookshelfLocation: Column<UpBookshelfRow>;
    Inventory: Column<UpBookshelfRow>;
    OperateUserName: Column<UpBookshelfRow>;
    Remark: Column<UpBookshelfRow>;
    CreateTime: Column<UpBookshelfRow>;
    UpdateTime: Column<UpBookshelfRow>;
}

export class UpBookshelfColumns extends ColumnsBase<UpBookshelfRow> {
    static readonly columnsKey = 'InventoryManage.UpBookshelf';
    static readonly Fields = fieldsProxy<UpBookshelfColumns>();
}